### ISSUE TITLE
PetaPOCO old version - Everything wrapped in a transaction?

### ISSUE THREAD

**Issue by picasso566** [CollaboratingPlatypus/PetaPoco#635]
On a set of projects, I'm implementing a DAL with PetaPoco V 5.1.244 (from 2017). 
_Note: I have a feeling that just updating everything to PetaPOCO latest and switching from the T4 to DBEntityGenerator will probably cure my ills, but this ticket is only for guidance. If anyone remembers if the older version of PetaPOCO wrapped calls in transactions automagically, it would be oh so helpful._ 

All of the apps connect to the same SQL server DB. I would like to set up temporal tables on the db and everything was testing OK until I ran into an issue:  Every update made with PetaPOCO causes 5-10 inserts into the temporal tables. The entries in the History have a valid period of 0 except the last one, which indicates the update was wrapped in a transaction with multiple updates. I am making heave use of the T4 generated wrappers as well.

Example: 
```
var cand = Candidate.Single(id);
cand.FirstInterviewDate =  DateTime.Now;
cand.Save();
```
This causes 5 History table inserts. For the life of me, I cannot see where a transaction is created in the code at all so I'm probably not looking close enough or missing something. 

Does anyone remember if this was a thing?
Does the latest version do this?

**Comment by asherber** [comment]
I don't think this was ever a thing in PetaPoco. (The main reason for the version jump from 5 to 6 was renaming of packages and deprecation of T4 templates when .NET Standard was added.) 

You can look at `database.LastCommand` after your `Save()` to see the SQL that was sent to the db, and you could also try trapping `TransactionStarted` and `CommandExecuting` to get a better sense of what's going on.

What do the History records look like? Does each one record a change to a single field in the record? If so, what happens when you do a multi-field update directly in SSMS?

**Comment by picasso566** [comment]
> 

Thanks for that. I can debug from here. Next step is profiling etc...
Direct updates in SSMS are fine. If I update 1 column or 5 there is a single history record inserted. 

If I do this through PP, 1 column or many, 5 records are inserted at the same time (validity period of 0 for the first 4 records). All of the history records records inserted are identical. It is very strange!

But from what you said and from my experience with PetaPOCO and stepping into each call in the PP library I don't see a transaction anywhere! So I was just double checking. My bet is I have c# code calling 5 updates.  This is also strange because I have traced the request lifecycle and I don't see the command running more than once in PP! It's an older project that needs a re-write at some point. Maybe I can convince the client that now is the time.
